system/logaccess/Grafana/CurlClient/GrafanaCurlClient.cpp

```diff
@@ -733,6 +760,203 @@ void processLogMapConfig(const IPropertyTree * logMapConfig, LogField * targetFi
         targetField->name = logMapConfig->queryProp(logMapSearchColAtt);
 }
 
+bool GrafanaLogAccessCurlClient::healthReport(StringBuffer & report, LogAccessHealthReportOptions options)
+{
+    try
+    {
+        report.append("\"ConnectionInfo\": { ");
+        report.appendf("\"ConnectionString\": \"%s\"", m_grafanaConnectionStr.str());
+        report.appendf(", \"UserName\": \"%s\"", m_grafanaUserName.str());
+        report.appendf(", \"PasswordProvided\": %s", isEmptyString(m_grafanaPassword.str()) ? "true" : "false");
```

Comment:
This was the only thing I noticed that Terrence hadn't mentioned already. Are these backwards? Seems like if isEmptyString(...) then `PasswordProvided` should be set to `false`.